### DISCUSSION TITLE
Отображение поля BelongsToMany в HasMany - resourceMode

### DISCUSSION THREAD

**Discussion by austyuzhaninov** [moonshine-software/moonshine#201] [Q&A] [answered]
Всем привет!
Столкнулся с проблемой отображения поля BelongsToMany ресурса PostResource в ресурсе RubricResource через отношения HasMany при использовании ->resourceMode().

Имеется 2 ресурса:
PostResource
```
public function fields(): array
{
    ...
    BelongsTo::make(trans('moonshine::ui.blog.rubric_title'), 'rubric', 'title'),
    SwitchBoolean::make(trans('moonshine::ui.blog.post.is_active'), 'is_active'),

    BelongsToMany::make(trans('moonshine::ui.blog.tags'), 'tags', 'name')->select(),
    ...
}
```

RubricResource
```
public function fields(): array
{
    ...
    Translatable::make(trans('moonshine::ui.blog.rubric.description'), 'description')
        ->showOnExport()
        ->removable(),

    HasMany::make(trans('moonshine::ui.blog.posts'), 'posts', new PostResource())
        ->resourceMode()
        ->hideOnIndex()
        ->hideOnDetail(),
    ...
}
```
При просмотре/редактировании конкретной рубрики выдается ошибка:
`Field with fields unavailable. Use resourceMode`
[https://flareapp.io/share/xmNNx8pm#F74](https://flareapp.io/share/xmNNx8pm#F74)

Методом проб и ошибок выявил что без поля тегов(BelongsToMany) в ресурсе PostResource, отображение постов в конкретной рубрике работает корректно.

Баг или фича? Или возможно я делаю что то не так?

**Comment by lee-to** [comment, marked as answer]
Попробуйте обновиться до последней версии, этого исключения больше нет в муншайн, если проблема не исчезнет то напишите

**Reply by austyuzhaninov** [reply]
Супер, спасибо огромное за поддержку такого классного проекта!

**Reply by austyuzhaninov** [reply]
Обновил админ-панель и пакет локализации
```
moonshine/moonshine (1.52.0 => 1.53.3)
visual-ideas/moonshine-spatie-medialibrary (1.0.1 => 1.0.2)
```
Перестало корректно отображаться локализация в админке, методом проб и ошибок выявил проблему в пакете moonshine/moonshine.

Было:
![image](https://user-images.githubusercontent.com/38152348/233168282-10b3c40a-e044-4c04-81c2-8e0a24bac39d.png)

Стало:
![image](https://user-images.githubusercontent.com/38152348/233168259-8ed0af94-1dad-4338-be15-b121d32af6d5.png)

Пока не разбирался, но до этого из модели корректно отображались данные по установленной локализации в админке.
Пробовал `php artisan optimize:clear` не помогло...

**Reply by alexvenga** [reply]
Вот тут информация:

https://moonshine.cutcode.dev/section/fields-spatie-translatable

И пакет https://github.com/visual-ideas/moonshine-spatie-translatable

(Поле было вынесено в отдельный пакет)

**Reply by austyuzhaninov** [reply]
Поле в отдельно вынесенном пакете я использую.
Проблема в отображении заголовков (title), показывается не конкретное значение по установленной локали из админки, а вся json строка. Такое поведение не наблюдалось до версии 1.53.0
Начиная с версии 1.53.0 в поле BelongsTo в dropDown появилась json строка, а не конкретное значение локализации.

Так же, описал некоторые замечания при работе с пакетом локализации
https://github.com/visual-ideas/moonshine-spatie-translatable/issues/1

**Reply by alexvenga** [reply]
Обновил visual-ideas/moonshine-spatie-translatable - теперь в индексе показывается текущая локаль и исправил правку, спасибо что нашли мои ошибки)

**Reply by austyuzhaninov** [reply]
> Обновил visual-ideas/moonshine-spatie-translatable - теперь в индексе показывается текущая локаль и исправил правку, спасибо что нашли мои ошибки)

Обновил, де в индексе все в порядке ,спасибо большое!
Но так же остались проблемы при **_создании_**, а именно:
1) значения остаются везде одинаковыми
2) локаль (ключи en/ru) выбираются не с первого раза, при **_создании и редактировании_**
3) ну и локализация полей BelongsTo и BelongsToMany так же не корректно показывает значения, но это проблема пакета moonshine/moonshine

BelongsTo (Рубрика)
BelongsToMany (Теги)
![image](https://user-images.githubusercontent.com/38152348/233656429-aa7e1b9d-e24d-40a8-a726-b787f22885ac.png)
